engine/include/Assets.h

```diff
@@ -114,6 +115,42 @@ namespace nc
         std::string leftPath;
     };
 
+    enum class DescriptorType : uint8_t
+    {
+        None,
+        UniformBuffer,
+        StorageBuffer,
+        CombinedImageSampler
+    };
+
+    enum class ShaderStages : uint8_t
+    {
+        None     = 0,
+        Vertex   = 1,
+        Fragment = 2
+    };
+
+    inline ShaderStages operator|(ShaderStages a, ShaderStages b)
+    {
+        return static_cast<ShaderStages>(static_cast<int>(a) | static_cast<int>(b));
+    }
+
+    struct DescriptorManifest
+    {
+        uint32_t setIndex;
+        uint32_t slotIndex;
+        DescriptorType descriptorType;
+        ShaderStages shaderStages;
+    };
+
+    struct ShaderView
```

Comment:
Using vectors instead of spans here makes this less of a 'view' and more of the actual asset data. Each copy of this will own its own memory, but the pattern for other assets is the actual memory is owned by the asset manager, and the views just 'point' to that data.

Also, do we need the string here?